proposals/3169/tags_facet.md

```diff
@@ -0,0 +1,259 @@
+    MODIFY COLUMN job_title
+    SET TAG visibility = 'public';
+```
+
+User-supplied tag through environment variables
+
+```bash
+    export OPENLINEAGE_TAG_ENVIRONMENT=prod
+```
+
+Run Event Example with Tags
+```json
+{
+        "eventType": "START",
+        "eventTime": "2020-12-28T19:52:00.001+10:00",
+        "run": {
+          "runId": "0176a8c2-fe01-7439-87e6-56a1a1b4029f",
+          "tags": [
+            {"key": "project", "value": "myproject", "source": "DBT_CLOUD_INTEGRATION"},
+            {"key": "environment", "value": "production", "source": "DBT_CLOUD_INTEGRATION"},
+            {"key": "crr_pipeline", "value": true, "source": "USER"},
+            {"key": "is_adhoc", "value": true, "source": "DBT_CLOUD_INTEGRATION"}
+            
+          ]
+        },
+        "job": {
+          "namespace": "my-namespace",
+          "name": "my-job",
+          "facets": {
+              "tags": [
+                {"key": "project", "value": "myproject", "source": "DBT_CLOUD_INTEGRATION"},
+                {"key": "environment", "value": "production", "source": "DBT_CLOUD_INTEGRATION"},
+                {"key": "crr_pipeline", "value": true, "source": "USER"}
+              ]
+          }
+        },
+        "inputs": [{
+          "namespace": "my-namespace",
+          "name": "my-input",
+          "facets": {
+            "tags": [
+                {"key": "meta_nr_config_slack_mentions": "value": "@user1, @user2", "source": "DBT_CLOUD_INTEGRATION"},
+                {"key": "pii", "value": true, "field": "email", "source": "DBT_CLOUD_INTEGRATION"}
+            ],
+            "schema": {
+                "fields": [
+                    {"name": "email", "type": "string" },
+                    {"name": "id", "type": "int"}
+                ] 
+                
+            }
+          }
+        }],  
+        "producer": "https://github.com/OpenLineage/OpenLineage/blob/v1-0-0/client",
+        "schemaURL": "https://openlineage.io/spec/1-0-5/OpenLineage.json#/definitions/RunEvent"
+      }
+```
+**Proposed implementation**
+
+This implementation proposes new tags facets that can be applied to jobs, runs, or datasets. The facet will have one property called tags. Tags is an array of objects holding key/value pairs and optional identifying information. While using key/value does not preserve the hierarchical nature of nested objects, it does allow us to collect all needed information supporting the vast majority of use cases.
+
+This schema will include a TagFacet with four properties. 
+* key: Identifies the tag by name
+* value: The value of the tag. Set to true to mimic single-string token. (pii=true)
+* source: (optional) Where the tag came from. This can be from the integration, a user, or the technology we are monitoring. 
+* field: (optional) Dataset field the tag applies to
+
+To add context to tags, we will generate TagJobFacet, TagRunFacet and TagDatasetFacet all referencing TagFacet.
```

Comment:
Thanks, I believe this addresses the concern I had - how to attach the same facet to different objects.